src/components/checkbox/checkbox.js

```diff
@@ -69,6 +80,7 @@ function MdCheckboxDirective(inputDirective, $mdAria, $mdConstant, $mdTheming, $
 
   function compile (tElement, tAttrs) {
     var container = tElement.children();
+    var indeterminateStateEnabled = tElement[0].hasAttribute('indeterminate');
```

Comment:
You can replace this one with `tAttrs.hasOwnProperty('indeterminate')` so you don't have to touch the DOM.